[Swift in WebKit] Work towards having a proper JSC module (part 2)

rr-codes opened the pull request:
#### 67be849c65f37affecc63563d12170d01ba7422a
<pre>
[Swift in WebKit] Work towards having a proper JSC module (part 2)
<a href="https://bugs.webkit.org/show_bug.cgi?id=304630">https://bugs.webkit.org/show_bug.cgi?id=304630</a>
<a href="https://rdar.apple.com/167064090">rdar://167064090</a>

Reviewed by Mark Lam.

Continue adding more missing header includes, this time focusing mostly on generated code.

* &lt;a lot of headers&gt;

Add missing include statements.

Canonical link: <a href="https://commits.webkit.org/304920@main">https://commits.webkit.org/304920@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/703a20ffabe2d9d8b548be37c1c68019af810646

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/136889 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/9249 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/48176 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/144625 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/89862 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/07f0bda8-3905-4f5d-b5d6-34f31aba6bdc) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/138761 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/9952 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/9095 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/104672 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/76389 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/a073850d-3de5-420b-8471-d6adc9a71e60) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/139834 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/7282 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/122647 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/85510 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/0a1a4af1-0b88-44ff-875c-61ff6b135741) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/6919 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/4615 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/5213 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/128850 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/116250 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/40830 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/147382 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/135375 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/8932 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/41400 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/113029 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/8950 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/7503 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/113359 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28797 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/6841 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/118930 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/63125 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/8980 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/36984 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/168155 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/8701 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/72546 "Built successfully") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/43874 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/8921 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/8772 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

MenloDorian commented on Source/JavaScriptCore/bytecode/BytecodeBasicBlock.h:
```diff
@@ -29,6 +29,7 @@
 #include <JavaScriptCore/Opcode.h>
 #include <limits.h>
 #include <wtf/FastBitVector.h>
+#include <wtf/TZoneMalloc.h>
```
This should be automatically included by config.h.  Otherwise, you'll have to include it everywhere.

MenloDorian commented on Source/JavaScriptCore/dfg/DFGLiveCatchVariablePreservationPhase.h:
```diff
@@ -25,6 +25,8 @@
 
 #pragma once
 
+#include <wtf/Platform.h>
```
This should be automatically included by config.h.  Why are we adding these again?

MenloDorian reviewed (approved):
r=me.  This is fine to land, though I wonder if all these files really need to be Private visibility or if some (or a lot) of them can just be Project visibility.

rr-codes committed:
[Swift in WebKit] Work towards having a proper JSC module (part 2)
https://bugs.webkit.org/show_bug.cgi?id=304630
rdar://167064090

Reviewed by Mark Lam.

Continue adding more missing header includes, this time focusing mostly on generated code.

* <a lot of headers>

Add missing include statements.

Canonical link: https://commits.webkit.org/304920@main
